dffml/util/net.py

```diff
@@ -8,6 +8,23 @@
 from .os import chdir
 
 
+# Hash Helper Function
+def secure_hash(string, algorithm, encoding="yes") -> str:
+    if encoding == "yes":
+        string = string.encode("utf-8")
+    if algorithm == "sha256":
+        return hashlib.sha256(string).hexdigest()
+    elif algorithm == "sha384":
+        return hashlib.sha384(string).hexdigest()
+
+
+def insecure_hash(string, algorithm="md5", encoding="yes") -> str:
+    if encoding == "yes":
```

Comment:
In the case that the caller passes bytes, we don't want to try to re-encode.

Also, is `encoding` currently used by any callers? Would it make more sense to just have an `isinstance()` check?

```suggestion
    if encoding == "yes" and isinstance(string, str):
```

Reply:
Encoding argument removed, now isinstance is checked.